classes/Cart.php

```diff
@@ -3838,7 +3837,7 @@ public function getSummaryDetails($id_lang = null, $refresh = false)
 
             if ($cart_rule['gift_product']) {
                 foreach ($products as $key => &$product) {
-                    if (empty($product['gift']) && $product['id_product'] == $cart_rule['gift_product'] && $product['id_product_attribute'] == $cart_rule['gift_product_attribute']) {
+                    if (empty($product['is_gift']) && $product['id_product'] == $cart_rule['gift_product'] && $product['id_product_attribute'] == $cart_rule['gift_product_attribute']) {
```

Comment:
Isn't this a BC break?

Reply:
not 100 % sure, but it seems just a mistake. Everywhere else is "is_gift", thats why this if stmt seems to be impossible to break through.